File: backend/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using backend.Entities.Base;
using backend.Entities.Enum;

namespace backend.Entities;

public class Order : BaseEntity
{
    public required Guid UserId  { get; set; }
    public Guid? VoucherId { get; set; }
    public OrderStatusEnum  Status { get; set; }
    public double TotalPrice { get; set; }
    public double PaidPrice { get; set; }
    [MaxLength(20)]
    public required string PhoneNumber {get; set;}
    [MaxLength(255)]
    public required string Address {get; set;}

    public ICollection<OrderFigure> OrderFigures { get; set; } = new List<OrderFigure>();
    public Voucher Voucher { get; set; }
}

File: frontend/src/types/order.ts
import { FigureDetailResponse } from "./figure";

export interface Order {
  id: string;
  userID: string;
  voucherID?: string | null;
  status: string;
  price: number;
  priceDiscounted: number;
  phoneNumber: string;
  address: string;
  createdAt: string;
  createdBy: string;
  updatedAt: string;
  updatedBy: string;
  figures?: OrderFigure[];
}

export interface OrderFigure {
  id: string;
  orderID: string;
  figureID: string;
  userID: string;
  quantity: number;
  price: number;
  figure: FigureDetailResponse;
}

export interface OrderResponse {
  id: string;
  user: {
    id: string;
    email: string;
  };
  status: string;
  totalPrice: number;
  paidPrice: number;
  orderDate: string;
  deliveryDate: string;
  phoneNumber: string;
  address: string;
  orderFigures: {
    orderId: string;
    figureId: string;
    price: number;
    quantity: number;
  }[];
}

export enum OrderStatus {
  Pending = "Pending",
  Processing = "Processing",
  Shipping = "Shipping",
  Completed = "Completed",
  Cancelled = "Cancelled"
}
